source/common/common/key_value_store_base.cc

```diff
@@ -60,6 +65,9 @@ bool KeyValueStoreBase::parseContents(absl::string_view contents,
 void KeyValueStoreBase::addOrUpdate(absl::string_view key, absl::string_view value) {
   store_.erase(key);
   store_.emplace(key, value);
+  if (!flush_timer_->enabled()) {
+    flush();
+  }
 }
 
 void KeyValueStoreBase::remove(absl::string_view key) { store_.erase(key); }
```

Comment:
Should we add a flush here when the timer is not enabled, to match the behavior in addOrUpdate()?